tests/e2e_application_aware_resource_quota_test.go

```diff
@@ -1968,6 +1997,17 @@ func newTestPodWithAffinityForQuota(f *framework.Framework, name string, affinit
 					Name:      "pause",
 					Image:     "busybox",
 					Resources: v1.ResourceRequirements{},
+					SecurityContext: &v1.SecurityContext{
+						Privileged:               ptr.To(false),
+						AllowPrivilegeEscalation: ptr.To(false),
+						RunAsNonRoot:             ptr.To(true),
+						SeccompProfile: &v1.SeccompProfile{
+							Type: v1.SeccompProfileTypeRuntimeDefault,
+						},
+						Capabilities: &v1.Capabilities{
+							Drop: []v1.Capability{"ALL"},
+						},
```

Comment:
nit: WDYT about creating a helper function to reduce duplications? Something like `generateUnprivilegedSecurityContext()`?

Reply:
Hey, thanks for the review! I'm actually planning a follow-up PR for refactoring soon, and I'll definitely focus on improving the pod creation then.